src/main/java/life/qbic/io/PetabParser.java

```diff
@@ -0,0 +1,85 @@
+package life.qbic.io;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import life.qbic.model.petab.PetabMetadata;
+
+public class PetabParser {
+
+  private final String META_INFO_YAML = "metaInformation.yaml";
+
+  public PetabMetadata parse(String dataPath) {
+
+    File directory = new File(dataPath);
+    List<String> sourcePetabReferences = new ArrayList<>();
+
+    File yaml = findYaml(directory);
```

Comment:
NIT: Maybe having a distinct exit condition makes sense? If a yaml was provided but is empty, this currently will return a PetabMetadata object with empty metadata, which could be confusing 🤔

Reply:
Currently only parent references are stored in the metadata and those can be empty, which is checked in the respective command.